hub/core/meta/tensor_meta.py

```diff
@@ -64,6 +64,37 @@ def __init__(
 
         super().__init__()
 
+    def adapt(self, buffer: memoryview, shape: Tuple[int], dtype):
+        dtype = np.dtype(dtype)
+        if self.dtype and self.dtype != dtype.name:
+            arr = np.frombuffer(buffer, dtype=dtype)
+            if np.can_cast(dtype, self.dtype):
+                buffer = memoryview(np.cast[self.dtype](arr).tobytes())
+            elif (
+                self.dtype == "uint8"
```

Comment:
Are we sure we want to add this? I know it's mentioned in AL-1193 but seems like unusual behaviour and just that's just enforced for uint8.
I think it's better to always just upcast and that way our casting will always be consistent.

Reply:
@istranic  ^^

Reply:
I used the 0-255 in the uint8 case as just an example. Ideally, this should work in all cases where we can upcast if there's no information loss.

If implementing this for all/most cases is difficult, let's skip it. If users complain about this error, we can create another ticket in the future.